bedboss/refgenome_validator/refgenomevalidator.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+
+
+class Validator:
+    """
+    Initialize validator class
+
+    1. filter out incompatible genomes
+    2. for remaining, determine tiers of compatibility
+
+
+    """
+
+    def __init__(self, bedfiles, genome_models):
+        # TODO ensure these are lists
+
+        self.bedfiles = bedfiles
+        self.genome_models = genome_models
+
+        self.compatibility_matrix = pd.DataFrame(
+            index=self.bedfiles, columns=self.genome_models
+        )  # non mutually exclusive  validation, i.e.
+
+        self.predicted_parent = self.predict_parent()
+
+    def predict_parent(self):
+        """
+        Predicts from a mutually exclusive list of reference genomes
+
+        """
+
+        # For mutually exclusive items, we can quickly check, chrom sizes and chrom names and see if there are incompatible
+        # genome models and truncate the list at the very beginning
+
+        predicted_parents = []
+
+        for bed in self.bedfiles:
+            # We may be doing this somewhere else and should just pass this in...
+            bed_chrom_info = get_bed_chrom_info(bed)
+
+            for genome in self.genome_models:
+                exclude_result = self.compare_chrom_size(
+                    bed_chrom_info, genome.chrom_sizes
+                )
+                if not exclude_result:
+                    # keep the genome
+                    predicted_parents.append(genome.alias)
+                else:
+                    # Can we for sure exclude this? if so delete it
+                    # Do we need to deep copy this or will it cause issues?
+                    self.genome_models.remove(genome)
+
+        predicted_parent = "hg38"
+
+        return predicted_parent
+
+    def modify_compatibility_matrix(self):
+        """
+
+        modifys the data frame storing compatibility matrix of entries
+
+        """
+
+        pass
+
+    def compare_chrom_size(self, bed_chrom_sizes, genome_chrom_sizes):
```

Comment:
Layer 1: check chrom names

loop through genomeModels and do `check_names()`.
Q=query
M=model

you can count these 3 numbers:
Q&M
Q&!M
!Q&M




Layer 2: check chrom lengths
for ones that pass layer 1...
for chroms in the Q&M category...
`check_lengths()`

you'll get either:
- everything fits
- stuff goes beyond


Layer 3:
`check_exclusions()`

Reply:
your goal here is to build a dict, with all these useful bits of information, for each ref genome (for the given bedfile).


then there's another function that takes that dict and produces the final recommendation.